dockstore-integration-testing/src/test/java/io/dockstore/webservice/UserResourceIT.java

```diff
@@ -562,4 +562,71 @@ public void testSetUserPrivilege() {
+        io.dockstore.openapi.client.api.UsersApi adminApi = new io.dockstore.openapi.client.api.UsersApi(adminWebClient);
+        io.dockstore.openapi.client.api.UsersApi userApi = new io.dockstore.openapi.client.api.UsersApi(userWebClient);
+        io.dockstore.openapi.client.model.User admin = adminApi.getUser();
+        io.dockstore.openapi.client.model.Profile userProfile = userApi.getUser().getUserProfiles().get("github.com");
+
+        // Should add a test above this to check that the API call should pass once the admin tokens are up to date
+        // Testing that the updateLoggedInUserMetadata() should fail if GitHub tokens are expired or absent
+        testingPostgres.runUpdateStatement(String.format("DELETE FROM token WHERE userid = %d", admin.getId()));
+        try {
+            adminApi.updateLoggedInUserMetadata("github.com");
+            fail("API call should fail and throw an error when no GitHub tokens are found or if tokens are out of date");
+        } catch (io.dockstore.openapi.client.ApiException ex) {
+            assertEquals(HttpStatus.SC_FORBIDDEN, ex.getCode());
+        }
+
+        // DockstoreUser2's profile elements should be initially set to null since the GitHub metadata isn't synced yet
+        assertNull(userProfile.getEmail());
+        assertNull(userProfile.getAvatarURL());
+        assertNull(userProfile.getLocation());
+
+        // The API call updateUserMetadata() should not throw an error and exit if any users' tokens are out of date or absent
+        // Additionally, the API call should go through and sync DockstoreTestUser2's GitHub data
+        userApi.updateUserMetadata();
+
+        userProfile = userApi.getUser().getUserProfiles().get("github.com");
+        assertEquals("dockstore.test.user2@gmail.com", userProfile.getEmail());
+        assertEquals("https://avatars1.githubusercontent.com/u/17859829?v=4", userProfile.getAvatarURL());
+        assertEquals("Toronto", userProfile.getLocation());
+    }
+
+    /**
+     * Tests the endpoint while all users have no valid GitHub token and the caller also does not have a valid token
+     */
+    @Test
+    public void testUpdateUserMetadataWithoutTokens() {
+        io.dockstore.openapi.client.ApiClient adminWebClient = getOpenAPIWebClient(ADMIN_USERNAME, testingPostgres);
+        io.dockstore.openapi.client.ApiClient userWebClient = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        io.dockstore.openapi.client.api.UsersApi adminApi = new io.dockstore.openapi.client.api.UsersApi(adminWebClient);
+        io.dockstore.openapi.client.api.UsersApi userApi = new io.dockstore.openapi.client.api.UsersApi(userWebClient);
+        io.dockstore.openapi.client.model.User admin = adminApi.getUser();
+        io.dockstore.openapi.client.model.Profile userProfile = userApi.getUser().getUserProfiles().get("github.com");
+
+        // Delete all of the tokens for every user
+        testingPostgres.runUpdateStatement("DELETE FROM token");
```

Comment:
If you delete the Dockstore token, then all authenticated endpoints are going to fail; it shouldn't even get into `UserResource.updateUserMetadata`. So then you are testing that the method is not being called at all, which isn't that useful in this scenario.

Note: if you were debugging this and seeing the method get invoked and seeming to contradict my previous statement, that would have been because the token is cached, and so even after you delete the token, calls may work for a bit. But it's not deterministic and you shouldn't rely on it.

This is a long-winded way of saying you should just delete the GitHub and/or Google tokens, not the Dockstore tokens.

Reply:
Done. Dockstore tokens are now left alone so the authenticated endpoints can be called successfully.